packages/hash/frontend/src/lib/org.ts

```diff
@@ -51,26 +53,76 @@ export const constructMinimalOrg = (params: {
 };
 
 export type Org = MinimalOrg & {
-  members: (MinimalUser & { responsibility: string })[];
+  members: (User & { responsibility: string })[];
 };
 
 export const constructOrg = (params: {
   subgraph: Subgraph;
   orgEntityEditionId: EntityEditionId;
+  resolvedUsers?: Record<EntityId, User>;
+  resolvedOrgs?: Record<EntityId, Org>;
 }): Org => {
   const { subgraph, orgEntityEditionId } = params;
 
-  return {
-    ...constructMinimalOrg({
+  const resolvedUsers = params.resolvedUsers ?? {};
+  const resolvedOrgs = params.resolvedOrgs ?? {};
+
+  const orgMemberships = getIncomingLinksForEntityAtMoment(
+    subgraph,
+    orgEntityEditionId.baseId,
+    new Date(),
+  ).filter(
+    (linkEntity) =>
+      linkEntity.metadata.entityTypeId ===
+      types.linkEntityType.orgMembership.linkEntityTypeId,
+  );
+
+  const org = constructMinimalOrg({
+    subgraph,
+    orgEntityEditionId,
+  }) as Org;
+
+  // We add it to resolved orgs *before* fully creating so that when we're traversing we know
+  // we already encountered it and avoid infinite recursion
+  resolvedOrgs[org.entityEditionId.baseId] = org;
+
+  org.members = orgMemberships.map(({ metadata, properties }) => {
+    const responsibility: string = properties[
+      extractBaseUri(types.propertyType.responsibility.propertyTypeId)
+    ] as string;
+
+    if (!metadata.linkMetadata?.leftEntityId) {
+      throw new Error("Expected org membership to contain a left entity");
+    }
+    const userEntity = getLeftEntityForLinkEntityAtMoment(
       subgraph,
-      orgEntityEditionId,
-    }),
-    /**
-     * @todo implement members once we are able to fetch incoming links
-     *   see https://app.asana.com/0/1201095311341924/1203250435416416/f
-     */
-    members: [],
-  };
-};
+      metadata.editionId.baseId,
```

Comment:
Should this be the following?
```suggestion
      metadata.linkMetadata?.leftEntityId,
```
Otherwise we are getting the OrgMembership, right?

Reply:
Nevermind, this is correct as it's getting the left entity from the org. Nevermind me :))